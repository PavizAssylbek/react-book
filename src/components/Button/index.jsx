import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

const Button = props => {

  const { onClick, className, outline, children } = props

  return (
    <button
      onClick={onClick}
      className={classNames('button', className, {
        'button--outline': outline,
      })}>
      {children}
    </button>
  );
};

Button.propTypes = {
  onClick: PropTypes.func,
  classname: PropTypes.string,
  outline: PropTypes.bool
};

Button.defaultProps = {
  onClick: () => {},
  classname: "",
  outline: false
} 

export default Button;
