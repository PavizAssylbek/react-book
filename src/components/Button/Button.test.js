import React from 'react'
import Button from './index'

const onClick = () => {
  console.log("check function")
}

describe('should render Button component', () => {

  it('should contain function onClick={Func}', () => {
    const component = shallow(<Button onClick={onClick} />)
    expect(component).toMatchSnapshot()
  });
  
  it('should contain string', () => {
    const component = shallow(<Button className="props" />)
    expect(component).toMatchSnapshot()
  });

  it('should contain boolean', () => {
    const component = shallow(<Button  outline={true} />)
    expect(component).toMatchSnapshot()
  });

  it('click', () => {
    const mockCallBack = jest.fn()
    const component = shallow(<Button onClick={mockCallBack} />)
    expect(mockCallBack.mock.calls.length).toEqual(0)
    component.find('.button').simulate('click')
    expect(mockCallBack.mock.calls.length).toEqual(1)
  })
})