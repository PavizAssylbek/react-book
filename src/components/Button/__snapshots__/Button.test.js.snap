// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Button component should contain boolean 1`] = `ShallowWrapper {}`;

exports[`should render Button component should contain function onClick={Func} 1`] = `ShallowWrapper {}`;

exports[`should render Button component should contain string 1`] = `ShallowWrapper {}`;
